src/fluree/db/flake/flake_db.cljc

```diff
@@ -292,6 +293,131 @@
           (merge-flakes t-new all-flakes)
           (assoc :commit commit-metadata)))))
 
+(defn- component->user-value
+  "Convert an internal pattern component to user-readable format"
+  [db component context]
+  (cond
+    (where/unmatched-var? component)
+    (str (where/get-variable component))
+
+    (where/matched-iri? component)
+    (let [iri (where/get-iri component)]
+      ;; Try to find matching prefix in context
+      (if-let [[prefix ns-iri] (some (fn [[k v]]
+                                       (when (and (string? v) (str/starts-with? iri v))
+                                         [k v]))
+                                     context)]
+        (str prefix ":" (subs iri (count ns-iri)))
+        iri))
+
+    (where/matched-value? component)
+    (where/get-value component)
```

Comment:
This is lacking support for `t`, `datatype`, and `lang` matchers.

Reply:
addressed in b7bb65dbd4caa98ca0b2c331ffe4aeee1b62cef5 in upstream branch